test/v3/models/domain/TaxYearSpec.scala

```diff
@@ -38,6 +38,31 @@ class TaxYearSpec extends UnitSpec {
       }
     }
 
+    "constructed from an ISO date" should {
+      "be the expected year, taking into account the UK tax year start date" in {
+
+        def test(datesAndExpectedYears: Seq[(String, Int)]): Unit = {
```

Comment:
Would it be better for this to take a single `dateAndExpectedYear` and then to use a `foreach` on the `input`. Maybe makes for sense for each set of values in the List to be individually tested? What do you think?

Reply:
Separate PR may be better for this